app/models/concerns/fog_extensions/vsphere/server.rb

```diff
@@ -35,7 +35,10 @@ def scsi_controller_type
       end
 
       def select_nic(fog_nics, nic)
-        fog_nics.detect {|fn| fn.network == nic.compute_attributes['network']} # grab any nic on the same network
+        nic_attrs = nic.compute_attributes
+        match =   fog_nics.detect { |fn| fn.network == nic_attrs['network'] } # grab any nic on the same network
```

Comment:
Hm yeah, I see what you mean, based on @shlomizadok's data then it's already working for him as he selects network with ID `DPortGroup 2`, which is returned already in `fog_nics` so the extra network lookup wouldn't be required.

In your case @fherbert, you select network with id `DPT_MAIN_PROD_200`, but `fog_nics` returns `dvportgroup-87` so you look up the DPT name and find the dvportgroup ID.

Different network configurations in vSphere perhaps, or different vSphere versions?

If so, it feels like a reasonable change as it's a fallback, not changing default behaviour.